quakemigrate/signal/local_mag/amplitude.py

```diff
@@ -585,13 +590,11 @@ def _get_amplitude_windows(self, station, i, event, p_ttimes, s_ttimes,
         s_end = s_pick + event.marginal_window + s_ttimes[i] * fraction_tt + \
             self.signal_window
 
+        # Check for overlaps
         if s_start < p_end:
-            if p_end < s_pick:
-                windows = [[p_start, p_end],
-                           [p_end, s_end]]
-            else:
-                windows = [[p_start, s_pick - event.marginal_window / 2],
-                           [s_pick - event.marginal_window / 2, s_end]]
+            mid_time = p_end + (s_start - p_end) / 2
+            windows = [[p_start, mid_time],
+                       [mid_time, s_end]]
         else:
             windows = [[p_start, s_start],
```

Comment:
Is this intentional? Rather than `p_end`?

Reply:
Yep, no real reason not to extend the P window as far as possible. The windows thing is all quite ambiguous; there doesn't seem to be any consensus on a standard practice, particularly for P.